vispy/gloo/glir.py

```diff
@@ -259,19 +260,25 @@ def as_es2_command(command):
 class BaseGlirParser(object):
     """ Base clas for GLIR parsers that can be attached to a GLIR queue.
     """
-    
+
+    def __init__(self):
+        self.capabilities = dict(
```

Comment:
It just doesn't make sense to me--the capabilities of the GL context are determined by the context, not the GLIR parser. It would also help to shorten the nesting you mentioned earlier. 

That said, I don't mind merging it as-is if you think it will be too much work to move it.